docs/specs/buffer.md

```diff
@@ -18,6 +18,20 @@ Vector buffers MUST be instrumented for optimal observability and monitoring. Th
 
 ### Events
 
+#### `BufferCreated`
+
+*All buffers* MUST emit a `BufferCreated` event immediately upon creation.
+
+* Properties
+  * `max_size` - the maximum number of events or byte size of the buffer
```

Comment:
I'm not entirely on board with this. Below, we have separate properties for the initial number of events and bytes, but here is a single property that represents either one or the other, depending on context? Does there need to be two events, one for a bytes-limited buffer, and the other for an events-limited buffer?

Reply:
Ah, I can see the concern. Perhaps another alternative is having `max_size_bytes` and `max_size_events` properties and specifying that they are mutually exclusive but that seems awkward. Implementation-wise, I was planning to make this distinction with an enum hence the one `max_size`. By two events did you have something like `DiskBufferCreated` and `MemoryBufferCreated` in mind? I'm fine with that route if we're ok with some overlap.

Reply:
👍 I think publishing two separate metrics: `max_size_bytes` and `max_size_events` makes sense. I can see a future where both buffer types support both limits, but for now they'd be mutually exclusive. You could model that as one `BufferCreated` event where the two different limits are `Option`s.